bpf/bpf_host.c

```diff
@@ -1029,6 +1030,17 @@ int to_netdev(struct __ctx_buff *ctx __maybe_unused)
 	}
 #endif
 
+#if defined(ENABLE_EGRESS_GATEWAY)
+	/* If the packet is revnat'ed, skip the egress gateway. */
+	if ((ctx->mark & MARK_MAGIC_SNAT_DONE) != MARK_MAGIC_SNAT_DONE) {
+		ret = egress_nat_fwd(ctx);
+		if (IS_ERR(ret))
+			return send_drop_notify_error(ctx, 0, ret,
+						      CTX_ACT_DROP,
+						      METRIC_EGRESS);
+	}
+#endif
```

Comment:
This block of code is causing the program size issue when combined with other features used in this file (bandwidth manager, BPF NodePort, host firewall, etc.). Maybe we could tail call before calling `snat_v4_process` (it depends whether we need to "come back" to this function to execute `lb_handle_health` below)?